src/components/application_manager/src/commands/mobile/set_audio_streaming_indicator_request.cc

```diff
@@ -0,0 +1,124 @@
+ may be used to endorse or promote products derived from this software
+ without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "application_manager/commands/mobile/set_audio_streaming_indicator_request.h"
+
+#include "application_manager/application.h"
+#include "application_manager/message_helper.h"
+
+namespace application_manager {
+
+namespace commands {
+
+SetAudioStreamingIndicatorRequest::SetAudioStreamingIndicatorRequest(
+    const MessageSharedPtr& message, ApplicationManager& application_manager)
+    : CommandRequestImpl(message, application_manager) {}
+
+SetAudioStreamingIndicatorRequest::~SetAudioStreamingIndicatorRequest() {}
+
+void SetAudioStreamingIndicatorRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+  if (!app) {
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    LOG4CXX_ERROR(logger_, "Application is not registered");
+    return;
+  }
+
+  if (!app->is_media_application()) {
+    LOG4CXX_INFO(logger_, "Application is not media");
+    SendResponse(false, mobile_apis::Result::REJECTED);
+    return;
+  }
+
+  if (app->audio_streaming_indicator() ==
+      static_cast<mobile_apis::AudioStreamingIndicator::eType>(
+          (*message_)[strings::msg_params][strings::audio_streaming_indicator]
+              .asInt())) {
+    LOG4CXX_INFO(logger_, "Application is subscribed to this indicator");
+    SendResponse(false, mobile_apis::Result::IGNORED);
+    return;
+  }
+
+  smart_objects::SmartObject msg_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  msg_params = (*message_)[strings::msg_params];
+
+  SendHMIRequest(
+      hmi_apis::FunctionID::UI_SetAudioStreamingIndicator, &msg_params, true);
+}
+
+void SetAudioStreamingIndicatorRequest::on_event(
+    const event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  switch (event.id()) {
+    case hmi_apis::FunctionID::UI_SetAudioStreamingIndicator: {
+      LOG4CXX_INFO(logger_, "Received UI_SetAudioStreamingIndicator event");
```

Comment:
@dtrunov complicated logic in switch is confusing, Please Extract some functionality to private functions.